sympy/physics/continuum_mechanics/truss.py

```diff
@@ -0,0 +1,494 @@
+        >>> t.loads
+        {'A': [[5*P/4, 90], [P/2, 45]], 'B': [[0, 90]]}
+        """
+        magnitude = sympify(magnitude)
+        direction = sympify(direction)
+
+        if location not in self.node_labels:
+            raise ValueError("Load must be applied at a known node")
+
+        else:
+            if direction in [self._loads[location][i][1] for i in range(len(self._loads[location]))]:
+                for i in range(len(self._loads[location])):
+                    if self._loads[location][i][1] == direction:
+                        self._loads[location][i][0] += magnitude
+                        break
+            else:
+                self._loads[location].append([magnitude, direction])
+
+    def add_support(self, location, type):
+        """
+        This method adds a pinned or roller support at a particular node
+
+        Parameters
+        ==========
+
+        location: String or Symbol
+            Label of the Node at which support is added.
+
+        type: String
+            Type of the support being provided at the node.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.add_support('A', 'pinned')
+        >>> t.supports
+        {'A': 'pinned', 'B': 'none'}
+        """
+        if location not in self._node_labels:
+            raise ValueError("Support must be added on a known node")
+
+        else:
+            self._supports[location] = type
+
+        if type == "pinned":
+            self._reaction_forces['R_'+str(location)+'_x'] = 0
+            self._reaction_forces['R_'+str(location)+'_y'] = 0
+
+        elif type == "roller":
+            self._reaction_forces['R_'+str(location)+'_y'] = 0
+            if 'R_'+str(location)+'_x' in list(self._reaction_forces):
+                self._reaction_forces.pop('R_'+str(location)+'_x')
+
+    def remove_support(self, location):
+        """
+        This method removes support from a particular node
+
+        Parameters
+        ==========
+
+        location: String or Symbol
+            Label of the Node at which support is to be removed.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.add_support('A', 'pinned')
+        >>> t.supports
+        {'A': 'pinned', 'B': 'none'}
+        >>> t.remove_support('A')
+        >>> t.supports
+        {'A': 'none', 'B': 'none'}
+        """
+        if location not in self._node_labels:
+            raise ValueError("No such node exists in the Truss")
+
+        else:
+            self._reaction_forces.pop('R_'+str(location)+'_y')
+            if self._supports[location] == "pinned":
+                self._reaction_forces.pop('R_'+str(location)+'_x')
```

Comment:
Similarly for removing a support, you should also remove the load you will be applying